sdk/servicebus/azure-servicebus/azure/servicebus/amqp/_amqp_message.py

```diff
@@ -123,7 +136,7 @@ def __init__(self, **kwargs):
             return
 
         # manually constructed AMQPAnnotatedMessage
-        input_count_validation = len([key for key in ("data_body", "sequence_body", "value_body") if key in kwargs])
+        input_count_validation = len([key for key in (data_body, sequence_body, value_body) if key])
```

Comment:
These cannot be included as explicit kwargs, b/c in the case that the user passes `value_body=None`, this will detect count as 0. @yunhaoling - do you know if there's a way to check that the user passed in `value_body=None`? or if no bodies are passed in, can we set `value_body` to `None` by default? and then validate that `input_count_validation <=1`?

Reply:
interesting.. I see what you mean that we cannot detect whether it's a None passed by user or default None.

if we set value_body to None by default, from a user perspective, previously they have to explicitly pass something, while now they can create an empty message with value_body being None.

bascially:

`msg = AmqpAnnotatedMessage(value_body=None)`
vs
`msg = AmqpAnnotatedMessage()`

and it shouldn't be a breaking changing to users, right? I feel I'm onboard with this idea but could you double check that this won't be a breaking change

Reply:
Here are the cases in which there would be a difference:
1) 
```python
AmqpAnnotatedMessage(...) # no bodies passed in
```
- Before: results `ValueError` b/c less than 1 body passed in.
- Now: created Message with value_body=None, without ValueError

2)
```python
AmqpAnnotatedMessage(data_body=..., value_body=...) # two bodies passed in, one is value_body
AmqpAnnotatedMessage(sequence_body=..., value_body=...) # two bodies passed in, one is value_body
```
- Before: results in ValueError since >1 bodies passed in
- Now: No ValueError. Message gets created with sequence_body

So the only difference is that what previously resulted in ValueError no longer does in these cases. It still does if multiple bodies are passed in and none of them are set to `None`.

Reply:
the second situation is a bit tricky, previously we have a stronger constraint by checking if users have passed a key; however now users can do 

`AmqpAnnotatedMessage(value_body=None, sequence_body=[xxxxx])` which is valid, and we can't tell whether users want to send a message with the value_body None or the sequence_body..

let's do not expand kwargs for those bodies. I'd rather not to introduce behavior changes just for the sake of making type hint look better.

Reply:
yep that makes more sense to me too